synapse/storage/databases/main/events_worker.py

```diff
@@ -1107,6 +1115,14 @@ async def _get_events_from_db(
             events_to_fetch = redaction_ids.difference(fetched_event_ids)
             if events_to_fetch:
                 logger.debug("Also fetching redaction events %s", events_to_fetch)
+                # Start tracing how long it takes for us to get all of the redactions
+                if not is_recording_redaction_trace:
+                    fetching_redactions_tracing_span_cm.__enter__()
+                    is_recording_redaction_trace = True
+
+        # Only stop recording if we were recording in the first place
+        if is_recording_redaction_trace:
+            fetching_redactions_tracing_span_cm.__exit__(None, None, None)
```

Comment:
If we're calling `__enter__` and `__exit__` manually, we probably want to ensure we call `__exit__` when there's an exception.

Reply:
I've refactored this code so we can use the normal `with` pattern. Please double check to make sure it's the same.